sync/course-db.js

```diff
@@ -656,11 +656,12 @@ module.exports.loadInfoFile = async function ({
     }
 
     // Validate file against schema
+    /** @type {import('ajv').ValidateFunction<T>} */
     const validate = ajv.compile(schema);
     try {
       const valid = validate(json);
       if (!valid) {
-        const result = { uuid: json.uuid };
+        const result = { uuid: /** @type {any} */ (json).uuid };
```

Comment:
TypeScript now [correctly propagates type guards that are extracted into constants](https://devblogs.microsoft.com/typescript/announcing-typescript-4-4/#cfa-aliased-conditions) like we have here - all the other changes in this file stemmed from that.